SimTracker/Configuration/python/SimTracker_EventContent_cff.py

```diff
@@ -62,3 +62,9 @@
         'keep *_*_Phase2OTDigiSimLink_*',
         'keep *_simSiPixelDigis_*_*', # covers digis and digiSimLinks
 ])
+
+from Configuration.Eras.Modifier_strips_vfp30_2016_cff import strips_vfp30_2016
+for _entry in [SimTrackerFEVTDEBUG,SimTrackerDEBUG,SimTrackerRAW,SimTrackerRECO,SimTrackerAOD,SimTrackerPREMIX]:
+    strips_vfp30_2016.toModify(_entry, outputCommands = _entry.outputCommands + [
+        'keep *_mix_SimulatedAPVDynamicGain_*'
```

Comment:
The `mix` label points to the `MixingModule` that does the digitization for the hard scatter event (~equivalent to the treatment of pileup in "stage 1"). Do you want to persist the product from that, or from the hard scatter + pileup overlay? For the latter the label is `mixData`.

(Naively I'd expect you'd want `mixData` here, and add `includeAPVSimulation=False` to
https://github.com/cms-sw/cmssw/blob/09562bc09c37b7207bb3f956b2834d9903572ea0/SimGeneral/MixingModule/python/stripDigitizer_cfi.py#L16-L26
In retrospect a comment along "these customizations should be replicated into stripDigitizer_cfi " in `SiStripSimParameters_cfi.py` above the `premix_stage1.toModify()` would have provided at least a reminder, even though a less error prone mechanism would be much better #26001.)

Reply:
@makortel, thanks for the comment, but please bear with me, I don't understand the suggestion in this jungle of configuration files.
* in the premixing case the APV simulation should be run *only* in the stage2. Why shall I set it false in `stripDigitizer_cfi`? I actually **want** it to be run in the stage2
* about `mix` or `mixdata`, as the APV simulation is run only in the stage2 I think we should persist the product from the hard scatter + pileup overlay. On the other hand it's a pity that the the name of the module is different in the premix case as now I need to speciate it for the premix case.

Reply:
> please bear with me, I don't understand the suggestion in this jungle of configuration files.

No problem, it's a bit complicated.

> * in the premixing case the APV simulation should be run _only_ in the stage2. Why shall I set it false in `stripDigitizer_cfi`? I actually **want** it to be run in the stage2

Premixing has three "mixing" actors
* in "stage 1" job (producing the pileup library)
  * 1: `MixingModule` to accumulate the data for pileup (label `mix`)
* in "stage 2" job (overlaying the pileup over signal
  * 2: `MixingModule` to accumulate the data for signal (label `mix`)
  * 3: `PreMixingModule` to overlay the pileup on top of signal (label `mixData`)

The two `MixingModule` instances need to be configured as for "stage 1". For `MixingModule` 1 the customization is in 
https://github.com/cms-sw/cmssw/blob/09562bc09c37b7207bb3f956b2834d9903572ea0/SimGeneral/MixingModule/python/SiStripSimParameters_cfi.py#L130-L134
and for `MixingModule` 2 the customization is in 
https://github.com/cms-sw/cmssw/blob/09562bc09c37b7207bb3f956b2834d9903572ea0/SimGeneral/MixingModule/python/stripDigitizer_cfi.py#L22-L26 
(actually one option to simplify would be to move also the former customization into `stripDigitizer_cfi`)

The `PremixingModule` (3) is then configured as for "stage 2", and there the `SiStripSimBlock` is used as-is
https://github.com/cms-sw/cmssw/blob/09562bc09c37b7207bb3f956b2834d9903572ea0/SimGeneral/PreMixingModule/python/mixOne_premix_on_sim_cfi.py#L62-L64

> On the other hand it's a pity that the the name of the module is different in the premix case as now I need to speciate it for the premix case.

I fully agree...

Reply:
>  On the other hand it's a pity that the the name of the module is different in the premix case as now I need to speciate it for the premix case.

An EDAlias can be defined to resolve this problem

Reply:
@makortel, thanks for the detailed explanation, it's much clearer now. I've implemented part of the suggestion in https://github.com/cms-sw/cmssw/pull/28355/commits/1e3c88cee5fda75042ba742fda32bd4c1af16fa3.

@slava77 I am afraid I am not really familiar with the `EDAlias`-ing. Shall an appropriate line be added here:
https://github.com/cms-sw/cmssw/blob/e626860d26692de5880c52c7c80aec7b859a0c60/SimGeneral/MixingModule/python/aliases_PreMix_cfi.py#L36
?

Reply:
> Shall an appropriate line be added here:
> https://github.com/cms-sw/cmssw/blob/e626860d26692de5880c52c7c80aec7b859a0c60/SimGeneral/MixingModule/python/aliases_PreMix_cfi.py#L36
> 
> 
> ?

I'd advise against using the (otherwise natural choice of) `simSiStripDigis` because I'm afraid that would make the already-confusing product naming even more confusing: as of today the `simSiStripDigis` is an alias for products of `mix`, i.e. from the "hard-scatter `MixingModule`". (and yes, the aliases really should be restructured such that downstream would see only `simSiStripDigis` for both classical mixing and premixing).

You could add a new alias though such that in `SimGeneral/MixingModule/python/aliases_cfi.py` the alias points to product from `mix` and in `SimGeneral/MixingModule/python/aliases_PreMix_cfi.py` it points to `mixData`. Then if/when #26001 gets resolved, the labels could be unified (resolving #26001 would anyway break nearly all product labels in DIGI).

Reply:
thanks @makortel I've tried my understanding of your proposed solution in this commit: https://github.com/cms-sw/cmssw/commit/435508be147e9ff70cf365d88be72dc1866e6f9e 
though while it seems to work fine for the standard mixing case, I don't see it working for the pre-mixing. Am I missing something trivial?

Reply:
actually, it seems that in the way it is configured now, the product is still coming from `mix` rather than `mixdata` (puzzled)